Yogic.py

```diff
@@ -1 +1,441 @@
+
+
+db.create_all()
+
+
+@app.route('/api/teacher', methods=['GET'])
+@jwt_required
+def get_teacher():
+    teacher_id=get_jwt_identity()
+    print('teacher id',teacher_id)
+    teacher = Teacher.query.filter_by(id=teacher_id['teacher']) #get the key from the get_jwt_identity
+    list_teacher = []
+    for _teacher in teacher:
+        d_teacher = dict()
+        d_teacher.update({'first_name': _teacher.first_name})
+        d_teacher.update({'last_name': _teacher.last_name})
+        d_teacher.update({'default_timezone': _teacher.default_timezone})
+        d_teacher.update({'years_experience': _teacher.years_experience})
+        d_teacher.update({'id': _teacher.id})
+        d_teacher.update({'email': _teacher.email})
+        list_teacher.append(d_teacher)
+        #print(type(_teacher))
+    print(json.dumps(list_teacher))
+    return Response(json.dumps(list_teacher), mimetype='application/json')
+
+
+#sign up form in react will call this method
+@app.route('/api/teacher', methods=['POST'])
+
+def create_teacher():
+    for key, value in request.headers.items():
+        print(key, value)
+    data = request.json
+    print(data)
+    new_teacher = Teacher(
+        first_name = data.get('first_name',''),
+        last_name = data.get('last_name',''),
+        email = data.get('email','').lower(),
+        years_experience =  int(data.get('years_experience',0)),
+        default_timezone =  data.get('default_timezone',''),
+        password_hash = set_password(data.get('email','').lower() + data.get('password', '')),
+    )
+    db.session.add(new_teacher)
+    try :
+        db.session.commit()
+        teacher_id = (new_teacher.id)
+        response = Response(json.dumps({'id':teacher_id}))
+    except:
+        response = Response(json.dumps({'error': 'Something went wrong,try again later'}))
+    response.headers['Content-type'] = 'application/json'
+    return response
+
+
+
+@app.route('/api/login/teacher', methods=['POST'])
+def login_teacher():
+    email = request.json.get('email','').lower()
+    password = request.json.get('password',None)
+    if not email or not password:
+        return jsonify({"msg": "Email or password invalid"}), 400
+
+
+    user = Teacher.query.filter_by(email=email).first()
+    if check_password(user.password_hash, email + password):
+        response = Response(json.dumps({'id':user.id}))
+        access_token = create_access_token(identity={'teacher':user.id})
+        response = Response(json.dumps({'access_token':access_token}))
+        response.headers['Content-type'] = 'application/json'
+        return response
+    else:
+        response=Response(json.dumps('wrong email or password'))
+        response.headers['Content-type'] = 'application/json'
+        return response
+
+
+
+@app.route('/api/teacher', methods=['PUT'])
+@jwt_required
+def edit_teacher():
+    claim = get_jwt_identity()
+    data = request.json
+    teacher_id = claim.get('teacher')
```

Comment:
will return null if it's a student, so you should return an unauthorized error
also applies to get/edit teacher/student